### ISSUE TITLE
datashader not importable with dask 2022.5.1

### ISSUE THREAD

**Issue by hyamanieu** [holoviz/datashader#1088]
I have set up a fresh python 3.9 conda environment with the holoviz stack, and datashader cannot be imported. I get the error down below after running `import datashader`, which indicates an issue within dask.

**Solution**: I have downgraded dask from `2022.5.1` to `2022.2.1` and it now works.

#### ALL software version info

`conda list`:
```
# Name                    Version                   Build  Channel
argon2-cffi               21.3.0             pyhd3eb1b0_0
argon2-cffi-bindings      21.2.0           py39h2bbff1b_0
asttokens                 2.0.5              pyhd3eb1b0_0
attrs                     21.4.0             pyhd3eb1b0_0
backcall                  0.2.0              pyhd3eb1b0_0
beautifulsoup4            4.11.1           py39haa95532_0
blas                      1.0                         mkl
bleach                    4.1.0              pyhd3eb1b0_0
bokeh                     2.4.3                      py_0    bokeh
bottleneck                1.3.4            py39h080aedc_0
brotli                    1.0.9                ha925a31_2
brotlipy                  0.7.0           py39h2bbff1b_1003
ca-certificates           2022.4.26            haa95532_0
certifi                   2022.5.18.1      py39haa95532_0
cffi                      1.15.0           py39h2bbff1b_1
charset-normalizer        2.0.4              pyhd3eb1b0_0
click                     8.0.4            py39haa95532_0
cloudpickle               2.0.0              pyhd3eb1b0_0
colorama                  0.4.4              pyhd3eb1b0_0
colorcet                  3.0.0                      py_0    pyviz
cryptography              37.0.1           py39h21b164f_0
cycler                    0.11.0             pyhd3eb1b0_0
cytoolz                   0.11.0           py39h2bbff1b_0
dask                      2022.5.0         py39haa95532_0
dask-core                 2022.5.0         py39haa95532_0
datashader                0.14.0                     py_0    pyviz
datashape                 0.5.4            py39haa95532_1
debugpy                   1.5.1            py39hd77b12b_0
decorator                 5.1.1              pyhd3eb1b0_0
defusedxml                0.7.1              pyhd3eb1b0_0
distributed               2022.5.0         py39haa95532_0
entrypoints               0.4              py39haa95532_0
executing                 0.8.3              pyhd3eb1b0_0
fonttools                 4.25.0             pyhd3eb1b0_0
freetype                  2.10.4               hd328e21_0
fsspec                    2022.3.0         py39haa95532_0
heapdict                  1.0.1              pyhd3eb1b0_0
holoviews                 1.14.9                     py_0    pyviz
hvplot                    0.8.0                      py_0    pyviz
icc_rt                    2019.0.0             h0cc432a_1
icu                       58.2                 ha925a31_3
idna                      3.3                pyhd3eb1b0_0
importlib-metadata        4.11.3           py39haa95532_0
intel-openmp              2021.4.0          haa95532_3556
ipykernel                 6.9.1            py39haa95532_0
ipython                   8.3.0            py39haa95532_0
ipython_genutils          0.2.0              pyhd3eb1b0_1
jedi                      0.18.1           py39haa95532_1
jinja2                    3.0.3              pyhd3eb1b0_0
jpeg                      9e                   h2bbff1b_0
jsonschema                4.4.0            py39haa95532_0
jupyter_client            7.2.2            py39haa95532_0
jupyter_core              4.10.0           py39haa95532_0
jupyterlab_pygments       0.1.2                      py_0
kiwisolver                1.4.2            py39hd77b12b_0
libpng                    1.6.37               h2a8f88b_0
libtiff                   4.2.0                he0120a3_1
libwebp                   1.2.2                h2bbff1b_0
llvmlite                  0.38.0           py39h23ce68f_0
locket                    1.0.0            py39haa95532_0
lz4                       3.1.3            py39h2bbff1b_0
lz4-c                     1.9.3                h2bbff1b_1
markdown                  3.3.4            py39haa95532_0
markupsafe                2.0.1            py39h2bbff1b_0
matplotlib                3.5.1            py39haa95532_1
matplotlib-base           3.5.1            py39hd77b12b_1
matplotlib-inline         0.1.2              pyhd3eb1b0_2
mistune                   0.8.4           py39h2bbff1b_1000
mkl                       2021.4.0           haa95532_640
mkl-service               2.4.0            py39h2bbff1b_0
mkl_fft                   1.3.1            py39h277e83a_0
mkl_random                1.2.2            py39hf11a4ad_0
msgpack-python            1.0.3            py39h59b6b97_0
multipledispatch          0.6.0            py39haa95532_0
munkres                   1.1.4                      py_0
nbclient                  0.5.13           py39haa95532_0
nbconvert                 6.4.4            py39haa95532_0
nbformat                  5.3.0            py39haa95532_0
nest-asyncio              1.5.5            py39haa95532_0
notebook                  6.4.11           py39haa95532_0
numba                     0.55.1           py39hf11a4ad_0
numexpr                   2.8.1            py39hb80d3ca_0
numpy                     1.21.5           py39h7a0a035_2
numpy-base                1.21.5           py39hca35cd5_2
openssl                   1.1.1o               h2bbff1b_0
packaging                 21.3               pyhd3eb1b0_0
pandas                    1.4.2            py39hd77b12b_0
pandocfilters             1.5.0              pyhd3eb1b0_0
panel                     0.13.1                     py_0    pyviz
param                     1.12.1                     py_0    pyviz
parso                     0.8.3              pyhd3eb1b0_0
partd                     1.2.0              pyhd3eb1b0_1
pickleshare               0.7.5           pyhd3eb1b0_1003
pillow                    9.0.1            py39hdc2b20a_0
pip                       22.0.4                   pypi_0    pypi
plotly                    5.6.0              pyhd3eb1b0_0
prometheus_client         0.13.1             pyhd3eb1b0_0
prompt-toolkit            3.0.20             pyhd3eb1b0_0
psutil                    5.8.0            py39h2bbff1b_1
pure_eval                 0.2.2              pyhd3eb1b0_0
pycparser                 2.21               pyhd3eb1b0_0
pyct                      0.4.8                      py_0    pyviz
pyct-core                 0.4.8                      py_0    pyviz
pygments                  2.11.2             pyhd3eb1b0_0
pyopenssl                 22.0.0             pyhd3eb1b0_0
pyparsing                 3.0.4              pyhd3eb1b0_0
pyqt                      5.9.2            py39hd77b12b_6
pyrsistent                0.18.0           py39h196d8e1_0
pysocks                   1.7.1            py39haa95532_0
python                    3.9.12               h6244533_0
python-dateutil           2.8.2              pyhd3eb1b0_0
python-fastjsonschema     2.15.1             pyhd3eb1b0_0
pytz                      2021.3             pyhd3eb1b0_0
pyviz_comms               2.2.0                      py_0    pyviz
pywin32                   302              py39h2bbff1b_2
pywinpty                  2.0.2            py39h5da7b33_0
pyyaml                    6.0              py39h2bbff1b_1
pyzmq                     22.3.0           py39hd77b12b_2
qt                        5.9.7            vc14h73c81de_0
requests                  2.27.1             pyhd3eb1b0_0
scipy                     1.7.3            py39h0a974cb_0
send2trash                1.8.0              pyhd3eb1b0_1
setuptools                61.2.0           py39haa95532_0
sip                       4.19.13          py39hd77b12b_0
six                       1.16.0             pyhd3eb1b0_1
sortedcontainers          2.4.0              pyhd3eb1b0_0
soupsieve                 2.3.1              pyhd3eb1b0_0
sqlite                    3.38.3               h2bbff1b_0
stack_data                0.2.0              pyhd3eb1b0_0
tbb                       2021.5.0             h59b6b97_0
tblib                     1.7.0              pyhd3eb1b0_0
tenacity                  8.0.1            py39haa95532_0
terminado                 0.13.1           py39haa95532_0
testpath                  0.5.0              pyhd3eb1b0_0
tk                        8.6.11               h2bbff1b_1
toolz                     0.11.2             pyhd3eb1b0_0
tornado                   6.1              py39h2bbff1b_0
tqdm                      4.64.0           py39haa95532_0
traitlets                 5.1.1              pyhd3eb1b0_0
typing-extensions         4.1.1                hd3eb1b0_0
typing_extensions         4.1.1              pyh06a4308_0
tzdata                    2022a                hda174b7_0
urllib3                   1.26.9           py39haa95532_0
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wcwidth                   0.2.5              pyhd3eb1b0_0
webencodings              0.5.1            py39haa95532_1
wheel                     0.37.1             pyhd3eb1b0_0
win_inet_pton             1.1.0            py39haa95532_0
wincertstore              0.2              py39haa95532_2
winpty                    0.4.3                         4
xarray                    0.20.1             pyhd3eb1b0_1
xz                        5.2.5                h8cc25b3_1
yaml                      0.2.5                he774522_0
zict                      2.0.0              pyhd3eb1b0_0
zipp                      3.8.0            py39haa95532_0
zlib                      1.2.12               h8cc25b3_2
```

#### Complete, minimal, self-contained example code that reproduces the issue

```
import datashader

```

#### Stack traceback and/or browser JavaScript console output

```
~\Anaconda3\lib\site-packages\datashader\__init__.py in <module>
      6 __version__ = str(param.version.Version(fpath=__file__, archive_commit="$Format:%h$",reponame="datashader"))
      7 
----> 8 from .core import Canvas                                 # noqa (API import)
      9 from .reductions import *                                # noqa (API import)
     10 from .glyphs import Point                                # noqa (API import)

~\Anaconda3\lib\site-packages\datashader\core.py in <module>
      6 import numpy as np
      7 import pandas as pd
----> 8 import dask.dataframe as dd
      9 import dask.array as da
     10 from xarray import DataArray, Dataset

~\Anaconda3\lib\site-packages\dask\__init__.py in <module>
      1 from . import config, datasets
      2 from ._version import get_versions
----> 3 from .base import annotate, compute, is_dask_collection, optimize, persist, visualize
      4 from .core import istask
      5 from .delayed import delayed

~\Anaconda3\lib\site-packages\dask\base.py in <module>
     16 
     17 from packaging.version import parse as parse_version
---> 18 from tlz import curry, groupby, identity, merge
     19 from tlz.functoolz import Compose
     20 

~\Anaconda3\lib\site-packages\tlz\__init__.py in <module>
      7 """
      8 
----> 9 from . import _build_tlz

~\Anaconda3\lib\site-packages\tlz\_build_tlz.py in <module>
      1 import sys
      2 import types
----> 3 import toolz
      4 from importlib import import_module
      5 

~\Anaconda3\lib\site-packages\toolz\__init__.py in <module>
     20 from . import curried, sandbox
     21 
---> 22 functoolz._sigs.create_signature_registry()
     23 
     24 from ._version import get_versions

NameError: name 'functoolz' is not defined
```

**Comment by ianthomas23** [comment]
Hi @hyamanieu. Sorry it has taken us so long to respond.

I haven't been able to reproduce this. On `win-64` if I use a clean Python 3.9 environment and `conda install -c pyviz datashader==0.14.0` I get the following packages:
```
# Name                    Version                   Build  Channel
blas                      1.0                         mkl  
bokeh                     2.4.3            py39haa95532_0  
bottleneck                1.3.5            py39h080aedc_0  
brotlipy                  0.7.0           py39h2bbff1b_1003  
ca-certificates           2022.4.26            haa95532_0  
certifi                   2022.6.15        py39haa95532_0  
cffi                      1.15.0           py39h2bbff1b_1  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
click                     8.0.4            py39haa95532_0  
cloudpickle               2.0.0              pyhd3eb1b0_0  
colorama                  0.4.5            py39haa95532_0  
colorcet                  3.0.0                      py_0    pyviz
cryptography              37.0.1           py39h21b164f_0  
cytoolz                   0.11.0           py39h2bbff1b_0  
dask                      2022.5.0         py39haa95532_0  
dask-core                 2022.5.0         py39haa95532_0  
datashader                0.14.0                     py_0    pyviz
datashape                 0.5.4            py39haa95532_1  
distributed               2022.5.0         py39haa95532_0  
freetype                  2.10.4               hd328e21_0  
fsspec                    2022.3.0         py39haa95532_0  
heapdict                  1.0.1              pyhd3eb1b0_0  
icc_rt                    2019.0.0             h0cc432a_1  
idna                      3.3                pyhd3eb1b0_0  
intel-openmp              2021.4.0          haa95532_3556  
jinja2                    3.0.3              pyhd3eb1b0_0  
jpeg                      9e                   h2bbff1b_0  
libpng                    1.6.37               h2a8f88b_0  
libtiff                   4.2.0                he0120a3_1  
libwebp                   1.2.2                h2bbff1b_0  
llvmlite                  0.38.0           py39h23ce68f_0  
locket                    1.0.0            py39haa95532_0  
lz4                       3.1.3            py39h2bbff1b_0  
lz4-c                     1.9.3                h2bbff1b_1  
markupsafe                2.1.1            py39h2bbff1b_0  
mkl                       2021.4.0           haa95532_640  
mkl-service               2.4.0            py39h2bbff1b_0  
mkl_fft                   1.3.1            py39h277e83a_0  
mkl_random                1.2.2            py39hf11a4ad_0  
msgpack-python            1.0.3            py39h59b6b97_0  
multipledispatch          0.6.0            py39haa95532_0  
numba                     0.55.1           py39hf11a4ad_0  
numexpr                   2.8.3            py39hb80d3ca_0  
numpy                     1.21.5           py39h7a0a035_3  
numpy-base                1.21.5           py39hca35cd5_3  
openssl                   1.1.1q               h2bbff1b_0  
packaging                 21.3               pyhd3eb1b0_0  
pandas                    1.4.3            py39hd77b12b_0  
param                     1.12.2                     py_0    pyviz
partd                     1.2.0              pyhd3eb1b0_1  
pillow                    9.2.0            py39hdc2b20a_1  
pip                       22.1.2           py39haa95532_0  
psutil                    5.9.0            py39h2bbff1b_0  
pycparser                 2.21               pyhd3eb1b0_0  
pyct                      0.4.8                      py_0    pyviz
pyct-core                 0.4.8                      py_0    pyviz
pyopenssl                 22.0.0             pyhd3eb1b0_0  
pyparsing                 3.0.4              pyhd3eb1b0_0  
pysocks                   1.7.1            py39haa95532_0  
python                    3.9.12               h6244533_0  
python-dateutil           2.8.2              pyhd3eb1b0_0  
pytz                      2022.1           py39haa95532_0  
pyyaml                    6.0              py39h2bbff1b_1  
requests                  2.28.1           py39haa95532_0  
scipy                     1.7.3            py39h0a974cb_0  
setuptools                61.2.0           py39haa95532_0  
six                       1.16.0             pyhd3eb1b0_1  
sortedcontainers          2.4.0              pyhd3eb1b0_0  
sqlite                    3.38.5               h2bbff1b_0  
tbb                       2021.5.0             h59b6b97_0  
tblib                     1.7.0              pyhd3eb1b0_0  
tk                        8.6.12               h2bbff1b_0  
toolz                     0.11.2             pyhd3eb1b0_0  
tornado                   6.1              py39h2bbff1b_0  
typing-extensions         4.1.1                hd3eb1b0_0  
typing_extensions         4.1.1              pyh06a4308_0  
tzdata                    2022a                hda174b7_0  
urllib3                   1.26.9           py39haa95532_0  
vc                        14.2                 h21ff451_1  
vs2015_runtime            14.27.29016          h5e58377_2  
wheel                     0.37.1             pyhd3eb1b0_0  
win_inet_pton             1.1.0            py39haa95532_0  
wincertstore              0.2              py39haa95532_2  
xarray                    0.20.1             pyhd3eb1b0_1  
xz                        5.2.5                h8cc25b3_1  
yaml                      0.2.5                he774522_0  
zict                      2.1.0            py39haa95532_0  
zlib                      1.2.12               h8cc25b3_2  
zstd                      1.5.2                h19a0ad4_0  
```
so exactly the same `datashader`, `dask`, `dask-core`, `distributed` and `toolz` as you have. With these I can `import datashader` without any problem.

Is there something special about your hardware and/or is this still a problem for you?

**Comment by hyamanieu** [comment]
Hello,

I tried on Windows 10. I didn't have this error, but another one: impossible to access `https://conda.anaconda.org/pyviz/noarch/param-1.12.2-py_0.tar.bz2` . This look like an intermittent error from the anaconda server, so I wouldn't open an issue for that. However in my original config it was param 1.12.1, so imposing this version worked, and I could import datashader.

I tried on Ubuntu 2022.04 using WSL, it worked as well. I will close this error, I can't reproduce it anymore
